kedro-datasets/kedro_datasets/polars/polars_dataset.py

```diff
@@ -0,0 +1,272 @@
+            load_method = getattr(pl, f"scan_{self._file_format}", None)
+            return load_method(load_path, **self._load_args)
+
+        # For object storage, we use pyarrow for I/O:
+        fsspec.filesystem("s3")
+        dataset = ds.dataset(load_path, filesystem=self._fs, format=self._file_format)
+        return pl.scan_pyarrow_dataset(dataset)
+
+    def _save(self, data: Union[pl.DataFrame, pl.LazyFrame]) -> None:
+        save_path = get_filepath_str(self._get_save_path(), self._protocol)
+        if Path(save_path).is_dir():
+            raise DatasetError(
+                f"Saving {self.__class__.__name__} to a directory is not supported."
+            )
+
+        if "partition_cols" in self._save_args:
+            raise DatasetError(
+                f"{self.__class__.__name__} does not support save argument "
+                f"'partition_cols'. Please use 'kedro.io.PartitionedDataset' instead."
+            )
+
+        BytesIO()
+        collected_data = None
+        if isinstance(data, pl.LazyFrame):
+            collected_data = data.collect()
+        else:
+            collected_data = data
+
+        save_method = getattr(collected_data, f"write_{self._file_format}", None)
+        if save_method:
+            buf = BytesIO()
+            save_method(file=buf, **self._save_args)
+            with self._fs.open(save_path, mode="wb") as fs_file:
+                fs_file.write(buf.getvalue())
+                self._invalidate_cache()
+        else:
+            raise DatasetError(
+                f"Unable to retrieve 'polars.DataFrame.write_{self._file_format}' "
+                "method, please ensure that your 'file_format' parameter has been "
+                "defined correctly as per the Polars API"
+                "https://pola-rs.github.io/polars/py-polars/html/reference/dataframe/index.html"
+            )
+
+    def _exists(self) -> bool:
+        try:
+            load_path = get_filepath_str(self._get_load_path(), self._protocol)
+        except DatasetError:
+            return False
+
+        return self._fs.exists(load_path)
+
+    def _release(self) -> None:
+        super()._release()
+        self._invalidate_cache()
+
+    def _invalidate_cache(self) -> None:
+        """Invalidate underlying filesystem caches."""
+        filepath = get_filepath_str(self._filepath, self._protocol)
+        self._fs.invalidate_cache(filepath)
+
+
+_DEPRECATED_CLASSES = {
+    "PolarsDataSet": PolarsDataset,
+}
+
+
+def __getattr__(name):
+    if name in _DEPRECATED_CLASSES:
+        alias = _DEPRECATED_CLASSES[name]
+        warnings.warn(
+            f"{repr(name)} has been renamed to {repr(alias.__name__)}, "
+            f"and the alias will be removed in Kedro-Datasets 2.0.0",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return alias
+    raise AttributeError(f"module {repr(__name__)} has no attribute {repr(name)}")
```

Comment:
Double checking again that `PolarsDataSet` doesn't actually exist right? These deprecation warning were added to the older datasets that were originally named with the capital "s" in `DataSet`. Any new datasets should just use the `Dataset` ending and there's no need to add this deprecation warning.